src/datastore/sync_methods/inject.js

```diff
@@ -313,12 +313,10 @@ module.exports = function inject (resourceName, attrs, options) {
     definition.emit('DS.beforeInject', definition, attrs)
   }
 
-  // start the recursive injection of data
+  // start the recursive injection of data; this will also call
+  // `definition.handleChange` for each item
   injected = _inject.call(_this, definition, resource, attrs, options)
 
-  // collection was modified
```

Comment:
Here's the difference between the old and the new behaviour, as I understand it (hope I got this right).

Old:
* `DS#findAll` -> multiple individual `DS.change` events (one per instance).
* `DS#inject` with an array of attributes -> multiple individual events (one per instance), then a final event with an array of instances.

New:
* `DS#findAll` -> one event with all instances.
* `DS#inject` with an array of attributes -> one event with all instances.